chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -199,14 +200,28 @@ async def puzzle_solution_received(self, coin_spend: CoinSpend, in_transaction:
         coin_name = coin_spend.coin.name()
         puzzle: Program = Program.from_bytes(bytes(coin_spend.puzzle_reveal))
         solution: Program = Program.from_bytes(bytes(coin_spend.solution)).rest().rest().first().first()
-        matched, singleton_curried_args, curried_args = nft_puzzles.match_nft_puzzle(puzzle)
-        if matched:
-            (_, metadata, metadata_updater_puzzle_hash, inner_puzzle) = curried_args
-            params = singleton_curried_args.first()
-            self.log.info(f"found the info for NFT coin {coin_name} {inner_puzzle} {params}")
-            singleton_id = bytes32(params.rest().first().atom)
+        try:
+            uncurried_nft = UncurriedNFT.uncurry(puzzle)
+        except Exception:
+            # The parent is not an NFT which means we need to scrub all of its children from our DB
```

Comment:
I don't see a test covering this case

Reply:
After a deep look, I think this case is impossible. Since we already checked and successfully uncurried NFT before this point. I will just remove this exception section